controller/volume_controller.go

```diff
@@ -2412,12 +2412,13 @@ func (vc *VolumeController) ReconcileShareManagerState(volume *longhorn.Volume)
 	return nil
 }
 
-func (vc *VolumeController) createShareManagerForVolume(volume, image string) (*longhorn.ShareManager, error) {
+func (vc *VolumeController) createShareManagerForVolume(volume *longhorn.Volume, image string) (*longhorn.ShareManager, error) {
 	sm := &longhorn.ShareManager{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      volume,
-			Namespace: vc.namespace,
-			Labels:    types.GetShareManagerLabels(volume, image),
+			Name:            volume.Name,
+			Namespace:       vc.namespace,
+			Labels:          types.GetShareManagerLabels(volume.Name, image),
+			OwnerReferences: datastore.GetOwnerReferencesForVolume(volume),
```

Comment:
Have we set the cascading deletion for the volume object?

Reply:
We don't specify and therefore use the default option of orphan.
But we do have a deletion call as part of the volume deletion already, as well as a finalizer (already) on the volume object.

Same as with engine/replica, just making it consistent with the ownership reference.
So that in the case if there is a share manager crd without the associated volume for whatever reason the GC can clean it up.